### PR TITLE
DOC - Enable viewing dev and stable versions of package documentation

### LINKED ISSUE #203: DOC add stable/dev versions of the doc
As done in https://scikit-learn.org/dev/ for example

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
## Context of the PR

Currently, it is impossible for a user to view the documentation of the stable version 0.3.
As we keep updating the package (new features, bugs, ...) the dev version of the doc diverges from the stable one.

A brief preview of this PR contribution
  
preview

Close #203 

## Contributions of the PR

- add a version toggler to switch between the dev and stable versions of the package documentation

### Checks before merging PR

- ~[ ] added documentation for any new feature~
- ~[ ] added unittests~
- ~[ ] edited the [what's new](../doc/changes/whats_new.rst)~

### CODE REVIEW

**Comment by mathurinm** [comment]
Thanks @Badr-MOUFAD ! On the CI build I get an error for https://output.circle-artifacts.com/output/job/724f72b1-185b-41e9-93cb-680e9da6ba85/artifacts/0/dev/stable/index.html when clicking on "stable version", is it normal ?

**Comment by Badr-MOUFAD** [comment]
> Thanks @Badr-MOUFAD ! On the CI build I get an error for https://output.circle-artifacts.com/output/job/724f72b1-185b-41e9-93cb-680e9da6ba85/artifacts/0/dev/stable/index.html when clicking on "stable version", is it normal ?

Yes @mathurinm, it is normal to not see the stable version.
We haven't uploaded it yet, and if so, it won't appear in the artifacts as we expose only the dev version in Circleci.

**Comment by mathurinm** [comment]
How do you think we should test this? Do you add the `stable` folder in this PR?

**Comment by Badr-MOUFAD** [comment]
> How do you think we should test this? Do you add the stable folder in this PR?

I exposed the stable doc [skglm v0.3](https://github.com/scikit-learn-contrib/skglm/tree/562f42b94a1bae58e2c16ad56a0c28f3a9317548) in the artifact, (beforehand, I built it and pushing it to ``gh-pages``)
Now, we have a mirror of what you will be deployed to the [website](https://contrib.scikit-learn.org/skglm/)

File: doc/conf.py [hunk 1]
```diff
@@ -241,7 +246,32 @@
 #html_use_smartypants = True
 
 # Custom sidebar templates, maps document names to template names.
-#html_sidebars = {}
+html_sidebars = {
+    "**": [
+        "sidebar/brand.html",
+        "sidebar/search.html",
+        "sidebar/navigation.html",
+        "sidebar/version_toggler.html",  # version toggler template
```
Comment (mathurinm) [inline doc/conf.py:254]: @Badr-MOUFAD maybe move it above as it contains important information? otherwise looking great!

Comment (Badr-MOUFAD) [reply doc/conf.py:254]: Doing so, +small adjustments, we get this render

screenshot for render

Looks great @mathurinm, :+1: for this change

**Review by mathurinm — APPROVED** [review]

**Comment by mathurinm** [comment]
@PABannier this should fix the issue you had, thanks @Badr-MOUFAD !

**Comment by PABannier** [comment]
Thanks @Badr-MOUFAD !

**Pull request merged**